src/librbd/AsyncResizeRequest.cc

```diff
@@ -119,8 +119,7 @@ void AsyncResizeRequest::send_grow_object_map() {
   bool object_map_enabled = true;
   {
     RWLock::RLocker l(m_image_ctx.owner_lock);
-    RWLock::RLocker l2(m_image_ctx.md_lock);
-    if (m_image_ctx.object_map == NULL) {
+    if (!m_image_ctx.object_map.enabled()) {
```

Comment:
Yeah, I think it'd need a larger refactoring to be able to do that from all paths. I was thinking we could leave it as is for now, since it'll work for offline addition/removal of object map, and worry about online changes later.